sftp.c

```diff
@@ -2211,54 +2158,22 @@ interactive_loop(struct sftp_conn *conn, char *file1, char *file2)
 	interactive = !batchmode && isatty(STDIN_FILENO);
 	err = 0;
 
-#ifdef WINDOWS
-	/* Min buffer size allowed in Windows is 2*/
-	setvbuf(stdout, NULL, _IOLBF, 2);
-	
-	/* We do this only in interactive mode as we are unable to read files with UTF8 BOM */
-	if (interactive) {
-		setvbuf(infile, NULL, _IOLBF, 2);
-		_setmode(_fileno(stdin), O_U16TEXT); /* prepare for Unicode input */
-	}
-#else   /* !WINDOWS */
 	setvbuf(stdout, NULL, _IOLBF, 0);
 	setvbuf(infile, NULL, _IOLBF, 0);
-#endif   /* !WINDOWS */
-
 
 	for (;;) {
 		char *cp;
 
 		signal(SIGINT, SIG_IGN);
 
 		if (el == NULL) {
-#ifdef WINDOWS
-			/* fgets on Windows does not support Unicode input*/
-			if (interactive) {
-				wchar_t wcmd[2048];
-				printf("sftp> ");
-				if (fgetws(wcmd, sizeof(cmd)/sizeof(wchar_t), infile) == NULL) {
-					printf("\n");
-					break;
-				}
-				else {
-					char *pcmd = NULL;
-					if ((pcmd = utf16_to_utf8(wcmd)) == NULL)
-						fatal("failed to convert input arguments");
-					strcpy(cmd, pcmd);
-					free(pcmd);                    
-				}
-			} else if (fgets(cmd, sizeof(cmd), infile) == NULL) 
```

Comment:
If you look into implementation of w32_fgets, it only calls fgetws for console io, file io is still on fgets()